spec/auto-completer-spec.js

```diff
@@ -1,150 +1,205 @@
+      });
+    });
+
+    it('should return nothing if no editor service', () => {
+      provider.editorService = null;
+
+      waitsForPromise(async () => {
+        var result = await getCompletions();
+        expect(result.length).toBe(0);
+      });
+    });
+
+    it('should return nothing if editor service call fails', () => {
+      spyOn(provider.editorService, 'getTypeaheadCompletionsAtPosition')
+        .andReturn(false);
+
+      waitsForPromise(async () => {
+        var result = await getCompletions();
+        expect(result.length).toBe(0);
+      });
+    });
+
+    it('should suggest matching elements', () => {
+      editor.setCursorBufferPosition([LINE_AFTER_DOM_MODULE, 0]);
+      editor.insertText('<te');
+
+      waitsForPromise(async () => {
+        var result = await getCompletions();
+        expect(result.length).toBe(1);
+        expect(result[0]).toEqual({
+          snippet: '<test-element $1></test-element>$0',
+          displayText: '<test-element>',
+          description: 'test description',
+          descriptionMarkdown: 'test description',
+          type: 'class',
+          replacementPrefix: '<te'
+        });
+      });
+    });
+
+    describe('attributes', () => {
+      it('should suggest matching attributes', () => {
+        editor.setCursorBufferPosition([LINE_AFTER_DOM_MODULE, 0]);
+        editor.insertText('<test-element fo');
+
+        waitsForPromise(async () => {
+          var result = await getCompletions();
+          expect(result.length).toBe(1);
+          expect(result[0]).toEqual({
+            displayText: 'foo',
+            snippet: 'foo="${1:string}"',
+            replacementPrefix: 'fo',
+            type: 'property',
+            description: 'foo description',
+            leftLabel: 'string'
+          });
+        });
+      });
+
+      it('should produce text suggestions for bools', () => {
+        editor.setCursorBufferPosition([LINE_AFTER_DOM_MODULE, 0]);
+        editor.insertText('<test-element ba');
+
+        waitsForPromise(async () => {
+          var result = await getCompletions();
+          expect(result.length).toBe(1);
+          expect(result[0]).toEqual({
+            text: 'bar',
+            type: 'property',
+            description: 'bar description',
+            leftLabel: 'boolean'
+          });
+        });
+      });
+
+      it('should display attribute inheritance');
+
+      describe('values', () => {
+        it('should autocomplete non-notifying properties', () => {
+          editor.setCursorBufferPosition([LAST_LINE_IN_TEMPLATE, 0]);
+          editor.insertText('<child-element non-notifying-property="');
+          const position = editor.getLastCursor().getBufferPosition();
+          editor.insertText('"></child-element>');
+          editor.setCursorBufferPosition(position);
+
+          waitsForPromise(async () => {
+            var result = await getCompletions();
+            expect(result.length).toBe(2);
```

Comment:
I actually took them from #54 and maintained the same structure. Fine by me to transform them all into `deepEqual` 😄